### ISSUE TITLE
What can be responsible for subtle abortions in btrbk?

### ISSUE THREAD

**Issue by RomuloPBenedetti** [digint/btrbk#156]
I started to do archiving daily and backups hourly and have noticed some abortions that I can't explain as all targets involved are fstab mounted and were available, here is my conf file:

```
transaction_log            /var/log/btrbk.log

preserve_day_of_week       sunday
timestamp_format           long-iso

snapshot_preserve_min      7d

snapshot_preserve          24h 10d
target_preserve            168h 30d 12w 6m
archive_preserve           168h 30d 24w 6m

snapshot_dir               snapshots

volume /mnt/FS_TREE/
    subvolume                root
        target send-receive  /mnt/HDDs/backups/snapshots/desktop
```
and here is the relevant log slice cleaned with relevant info only:

```
-------------from Cron.Hourly -  Sun Jun 4 20:01:01 -03 2017  ------------
2017-06-04T20:01:11-0300 finished success 10 - - - -
-------------from Cron.Hourly -  Sun Jun 4 21:01:01 -03 2017  ------------
2017-06-04T21:01:02-0300 finished partial 0 - - - # At least one backup task aborted
.
.    all equaly aborted as Sun Jun 4 21:01:01 -03 2017
.
-------------from Cron.Hourly -  Sun Jun 4 21:01:01 -03 2017  ------------
2017-06-05T03:35:04-0300 abort_target ABORT - /mnt/BigHDD/backups/snapshots/desktop - - # Receive targets of archive candidates exist at unexpected location
2017-06-05T03:35:04-0300 abort_archive_source ABORT - /mnt/HDDs/backups/snapshots/desktop - - # At least one target aborted earlier
2017-06-05T03:35:04-0300 finished partial 3 - - - # At least one backup task aborted
-------------from Cron.Hourly -  Mon Jun 5 04:01:01 -03 2017  ------------
2017-06-05T04:01:01-0300 finished partial 0 - - - # At least one backup task aborted
.
.    all equaly aborted as Mon Jun 5 04:01:01 -03 2017
.
-------------from Cron.Hourly -  Mon Jun 5 09:01:01 -03 2017  ------------
2017-06-05T09:02:18-0300 finished success 76 - - - -
 RETURNS TO NORMAL SNAPSHOTING
```
Another point is that there exist many codes I cant explain as "success 76 or 10".

**Comment by digint** [comment]
```
2017-06-04T21:01:02-0300 finished partial 0 - - - # At least one backup task aborted
```
I don't have enough context to give an explanation here. Some output from "normal" log (non-transaction log) would help.

```
2017-06-05T03:35:04-0300 abort_target ABORT - /mnt/BigHDD/backups/snapshots/desktop - - # Receive targets of archive candidates exist at unexpected location
2017-06-05T03:35:04-0300 abort_archive_source ABORT - /mnt/HDDs/backups/snapshots/desktop - - # At least one target aborted earlier
2017-06-05T03:35:04-0300 finished partial 3 - - - # At least one backup task aborted
```
This is the result of a `btrbk archive` operation, where btrbk found subvolumes at "unexpected location", meaning that a subvolume to be transferred to the archive target is already present there, but not at the location btrbk expect it to (possibly from an older `btrbk archive` call with different directory or some manual action). While for normal `btrbk run` actions, "unexpected location" only emits a warning, it leads to abort for `btrbk archive`. (Note that this "unexpected location" behavior has already caused a lot of confusion and is subject to change).

Sidenote: are you sure you need to run `btrbk archive` in a cron-job? This command is meant to be run manually in order to "archive" disks. Maybe what you actualy want is something like this:

```
volume /mnt/FS_TREE/
    subvolume                root
        target send-receive  /mnt/HDDs/backups/snapshots/desktop
          target_preserve    168h 30d 12w 6m
        target send-receive  /mnt/<my-second-bigger-disk>
          target_preserve    168h 30d 24w 6m
```


> Another point is that there exist many codes I was not able to find a description for as "success 76 or 10".

It's the total duration of the operation (see btrbk.conf(5), transaction_log: "...in a space-separated table format: "localtime type status *duration* target_url source_url parent_url message".)

**Comment by RomuloPBenedetti** [comment]
> I don't have enough context to give an explanation here. Some output from "normal" log (non-transaction log) would help.

Well there is not much  extra info: 
![spectacle lh1233](https://user-images.githubusercontent.com/12467382/26861274-207940b0-4b19-11e7-8fa8-e5c300d4a356.png)


> Sidenote: are you sure you need to run btrbk archive in a cron-job? This command is meant to be run manually in order to "archive" disks. Maybe what you actualy want is something like this:

I'm pretty much on an early implementation of the backup settings, but I'm only using archive because I need to store with different policies, it still is not there but I plan to have different targets policy, is it possible from what I understand on your example?

**Comment by digint** [comment]
Nope, that does not help. In your cron script, try running something like `btrbk run 2>>/tmp/btrbk.err` (or simply run `btrbk run` in a shell) to see the errors and warnings.

> I'm pretty much on an early implementation of the backup settings, but I'm only using archive because I want to store with different policies in future, it still is not there but I plan to have different targets policy, is it possible from what I understand on your example?

Yes, all options can be set in "global", "volume", "subvolume" or "target" sections of the config file, making the preserve policies very versatile.

**Comment by RomuloPBenedetti** [comment]
Unfortunately it will show the same message when it fails because my cronjob as you can see do run without `--quiet` or anything like that. Anyway, I believe I discovered the culprit at last, apparently **systemd**'s fstab `x-systemd.automount` option appear to not mount some partitions at the end of boot and stay like so until you try to access or `ls` the partition.

Thanks for the clarification, I've adjusted and now I only relies on targets.
